test/AxelarGateway.js

```diff
@@ -370,6 +503,18 @@ describe('AxelarGateway', () => {
             await expect(implementation.connect(governance).setup(params)).to.be.revertedWithCustomError(implementation, 'NotProxy');
         });
 
+        it('should not allow malicious proxy to call setup function directly and transfer governance or mint limiter', async () => {
```

Comment:
Simulates someone deploying their own instance of the gateway proxy, pointing it to the existing gateway implementation, and passing in setup params to transfer governance/mint limiter to a different address

Reply:
yeah, but i don't see setup etc. being called in the test?

Reply:
Since it's just another copy of AxelarGatewayProxy setup is called in the constructor when it's deployed. I could make a test proxy that calls setup more explicitly or remove the test as it doesn't add to test coverage at all. I just saw some people bring up this question in code4rena so I thought I might as well add a test for it.